### PR TITLE
Forward port of 7.0 to 8.0 with last pylint fixes

### PR DESCRIPTION
**Pull request by guewen**
I pulled https://github.com/OCA/e-commerce/pull/21  
So it includes all the last changes of 7.0 + the pylint fixes.                                                                                                                                                 

Also, .travis.yml conflicted so I corrected it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
-[![Build Status](https://travis-ci.org/OCA/e-commerce.svg?branch=master)](https://travis-ci.org/OCA/e-commerce)
-[![Coverage Status](https://img.shields.io/coveralls/OCA/e-commerce.svg)](https://coveralls.io/r/OCA/e-commerce?branch=master)
+[![Build Status](https://travis-ci.org/OCA/e-commerce.svg?branch=8.0)](https://travis-ci.org/OCA/e-commerce)
+[![Coverage Status](https://coveralls.io/repos/OCA/e-commerce/badge.png?branch=8.0)](https://coveralls.io/r/OCA/e-commerce?branch=7.0)
```
Comment (lmignon) [inline README.md:3]: branch=8.0

Comment (guewen) [inline README.md:3]: oops

File: __unported__/sale_automatic_workflow/stock.py [hunk 2]
```diff
@@ -54,6 +55,7 @@ def validate_picking(self, cr, uid, ids, context=None):
                 }
             self.do_partial(cr, uid, [picking.id], partial_data)
         return True
+<<<<<<< HEAD:__unported__/sale_automatic_workflow/stock.py
```
Comment (lmignon) [inline __unported__/sale_automatic_workflow/stock.py]: merge conflict

**Comment by lmignon** [comment]
Code LGTM, it only remains to correct Travis for tests.

**Comment by guewen** [comment]
@lmignon what do you mean?

edit: I got what you mean. The tests failure was due to a new module in v7, so I moved it in `__unported__`

**Comment by lmignon** [comment]
:+1: Thank you for the forward port.

**Comment by pedrobaeza** [comment]
:+1: 

I'm seeing this forward ports maybe unneeded when there is no module still migrated, because I always take last 7.0 version to start the migration. Don't you think?

**Comment by guewen** [comment]
@pedrobaeza the 8.0 branch was already there so I prefer to keep it

**Pull request merged**